### DISCUSSION TITLE
Fail on attempt to compile

### DISCUSSION THREAD

**Discussion by mhsiegel14** [iraf-community/iraf#168] [General]
I know next to nothing about how these things work.  But I'm getting an error when running the make sysgen step on a Mac with OSX Mojave. After a whole bunch of "file not found alerts", it crashes out with the message:

"ld: file too small (length=8) file '/users/mhs18/iraf/iraf/lib/libsys.a' for architecture x86_64"
"clang: error: linker command failed with exit code 1 (use -v to see invocation)"

It might be that I haven't fully upgraded to Catalina yet or something. I'm also having to run this from an admin account since i can't install software from my primary account. If anyone has any tips, I would appreciate it.

**Comment by olebole** [comment]
Can you attach the full log (`build.log` if you did it like described in the installation procdure)?

**Reply by mhsiegel14** [reply]
Sure. Attached. Thanks!
[build.log](https://github.com/iraf-community/iraf/files/6548367/build.log)

**Reply by olebole** [reply]
This looks like that you already have the `iraf` environment variable set. Can you check that with `echo $iraf`, and do an `unset iraf` before re-running `./install`?

**Reply by mhsiegel14** [reply]
That seemed to help a bit but it crashed again.
[build.log](https://github.com/iraf-community/iraf/files/6548834/build.log)

**Reply by olebole** [reply]
Did you ever try to install an "old" version of IRAF (from iraf.noao.edu)? In this case, a lot of environment variables are set which may disturb the building. The main problem starts here (line 1274 of your build log):
```
xc -c -w access.x aread.x areadb.x await.x awaitb.x […] zfiott.x
/bin/bash: xc: command not found
```
xc however was built and installed properly. build in lines 591-598:
```
----------------------- XC  ----------------------------
+ gcc -c -g -Wall -O2 -I/users/mhs18/iraf/iraf/include -DNOVOS xc.c
xc.c:1413:1: warning: unused function 'findexe' [-Wunused-function]
findexe (
^
1 warning generated.
+ gcc xc.o /users/mhs18/iraf/iraf/unix/hlib/libboot.a /users/mhs18/iraf/iraf/unix/hlib/libos.a -lm -o xc.e
+ mv -f xc.e ../../hlib
```
install in line 1268:
```
+ mv -f hlib/alloc.e hlib/generic.e hlib/mkpkg.e […] hlib/xc.e hlib/xpp.e hlib/xyacc.e bin.macintel
```
You should therefore see these binaries (`alloc.e`, …, `xc.e`) in `unix/bin.macintel`, and `unix/bin/` should be symlinked to  `unix/bin.macintel`, right? Also, the directory `${HOME}/.iraf/bin/` should contain symlinks into your new IRAF installation, like this:
```
$ ls -l ~/.iraf/bin
[…]
lrwxr-xr-x  1 mhs18  staff  38 26 Mai 14:03 xc -> /Users/mhs18/iraf/iraf/unix/bin.macintel/xc.e
```
If this is not the case, I would try to remove (or rename) the whole `${HOME}/.iraf/`  subdirectory (or at least the `.iraf/bin`) and then do again the `./install` (and the two `make`s).

**Reply by olebole** [reply]
Oh -- and did you check that you got an `[  OK  ]` on all steps when you ran `./install`? Maybe you can also provide the output of this?

**Reply by mhsiegel14** [reply]
Just an update. The installer did not have any errors. I moved the .iraf directories both in the account I'm targeting and in the admin account where I have to run the scripts from.  The symlinks all seem to be fine. But it keeps crashing out at the same place.
[build.log](https://github.com/iraf-community/iraf/files/6562685/build.log)

**Reply by olebole** [reply]
Did you check that `~/.iraf/bin/` contains the correct links?
```
$ ls -l ~/.iraf/bin
[…]
lrwxr-xr-x  1 mhs18  staff  38 26 Mai 14:03 xc -> /Users/mhs18/iraf/iraf/unix/bin.macintel/xc.e
```

**Reply by mhsiegel14** [reply]
I am wondering if I'm confusing the scripts by running this with an admin account and not the mhs18 account (university requires this).

**Reply by mhsiegel14** [reply]
Hold on.  DIdn't copy the output right 

bash-3.2$ ls -l /users/mhs18/.iraf/bin
total 0
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  38 May 28 14:04 cl -> /users/mhs18/iraf/iraf/unix/hlib/cl.sh
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  39 May 28 14:04 ecl -> /users/mhs18/iraf/iraf/unix/hlib/ecl.sh
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  50 May 28 14:04 generic -> /users/mhs18/iraf/iraf/unix/bin.macintel/generic.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  42 May 28 14:04 mkiraf -> /users/mhs18/iraf/iraf/unix/hlib/mkiraf.sh
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  43 May 28 14:04 mkmlist -> /users/mhs18/iraf/iraf/unix/hlib/mkmlist.sh
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  48 May 28 14:04 mkpkg -> /users/mhs18/iraf/iraf/unix/bin.macintel/mkpkg.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  48 May 28 14:04 rmbin -> /users/mhs18/iraf/iraf/unix/bin.macintel/rmbin.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  50 May 28 14:04 rmfiles -> /users/mhs18/iraf/iraf/unix/bin.macintel/rmfiles.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  46 May 28 14:04 rpp -> /users/mhs18/iraf/iraf/unix/bin.macintel/rpp.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  47 May 28 14:04 rtar -> /users/mhs18/iraf/iraf/unix/bin.macintel/rtar.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  54 May 28 14:04 sgidispatch -> /users/mhs18/iraf/iraf/unix/bin.macintel/sgidispatch.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  47 May 28 14:04 wtar -> /users/mhs18/iraf/iraf/unix/bin.macintel/wtar.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  45 May 28 14:04 xc -> /users/mhs18/iraf/iraf/unix/bin.macintel/xc.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  46 May 28 14:04 xpp -> /users/mhs18/iraf/iraf/unix/bin.macintel/xpp.e
lrwxr-xr-x  1 mhs18_admin  PSU\Domain Users  48 May 28 14:04 xyacc -> /users/mhs18/iraf/iraf/unix/bin.macintel/xyacc.e

**Reply by olebole** [reply]
Without knowing exactly what happens there, the following should work:
```
bash-3.2$ PATH="/users/mhs18/.iraf/bin:$PATH"
bash-3.2$ make sysgen 2>&1 | tee build.log
```
Could you try this? There seems something strange with setting the proper PATH.

**Reply by mhsiegel14** [reply]
Wow, that worked!  Thank you so much!

**Reply by mhsiegel14** [reply]
Hmm.  Runs in my admin account but the main gives me the error:
os.zgtenv: cannot follow link '/usr/include/iraf.h'
task 'cl' has no param file
Fatal startup error. CL dies

I'm guessing I've got some path set that works for admin but not for the main account. I'll see if I can figure it out.

**Reply by olebole** [reply]
I must say that I do not fully understand why you are fiddling here with two different accounts; this obviously make things complicated and causes problems. You can install IRAF without problems (and without admin rights) on your personal account.
I would recommend to at least copy the `~/.iraf` dir to your own home dir and adjust the file `login.cl` to your local home dir (especially the settings on top). And you should initialize this on login, so add these lines into your `~/.bashrc`:
```bash
# Add iraf setup commands
if [ -e ${HOME}/.iraf/setup.sh ]; then
    source ${HOME}/.iraf/setup.sh
fi
```
See the `.bashrc` of your admin account.

**Reply by mhsiegel14** [reply]
I agree it's silly but have little choice with the computing rules.  That fixed the cannot follow links part but still getting cl has no param file error. Not sure where it's choking.

**Reply by olebole** [reply]
Are you sure that the `iraf` environment variable is correctly set?

**Reply by mhsiegel14** [reply]
seems to be. $iraf references to the correct place.
